internal/wal_uploader.go

```diff
@@ -0,0 +1,64 @@
+package internal
+
+import (
+	"github.com/wal-g/storages/storage"
+	"github.com/wal-g/wal-g/internal/compression"
+	"github.com/wal-g/wal-g/utility"
+	"io"
+	"path"
+)
+
+// WalUploader extends uploader with wal specific functionality.
+type WalUploader struct {
+	*Uploader
+	*DeltaFileManager
+}
+
+func (uploader *WalUploader) getUseWalDelta() (useWalDelta bool) {
+	return uploader.DeltaFileManager != nil
+}
+
+func NewWalUploader(
+	compressor compression.Compressor,
+	uploadingLocation storage.Folder,
+	deltaFileManager *DeltaFileManager,
+) *WalUploader {
+	uploader := NewUploader(compressor, uploadingLocation)
+
+	return &WalUploader{
+		uploader,
+		deltaFileManager,
+	}
+}
+
+// Clone creates similar WalUploader with new WaitGroup
+func (uploader *WalUploader) clone() *WalUploader {
```

Comment:
I can't find any usage of this function. Delete if this function unnecessary. If it is using somewhere I should take a better look at this function, because it could be really dangerous to copy deltafilemanager

Reply:
It is used in bgUploader. It does not copy deltafilemanager, just returns the same one.